### DISCUSSION TITLE
I did parameterization. Is there any way I can only run some parameters in the parameter group when running a testcase

### DISCUSSION THREAD

**Discussion by wang1150297847** [pytest-dev/pytest#10858] [Q&A]
@pytest.mark.parametrize("case_lel,id,", [('P0',0),('P1',1),('P2',2)])
#demo.py
class TestClass_A:
    def test_demo(self,case_lel,id):
        print(id)

now:   pytest demo.py::TestClass_A::test_demo
it will run parameters ('P0',0) , ('P1',1) and ('P2',2)
can somtime  I run only one set of parameters？
eg：i just use  ('P0',0)  or i use ('P0',0) , ('P1',1)
can run like this ?  pytest demo.py::TestClass_A::test_demo::parametrize['P0','P1']
Or is there any other way to achieve the functions I need

**Comment by arthuRHD** [comment]
Hi @wang1150297847 ,

From what I understand, you need randomness in your parameterization.
```py
import pytest
import random

parameters = [(1,11),(2,22),(4,44)]

number_of_param_selected = random.randint(1, len(parameters))
randomized_parameters = random.choice(parameters, k=number_of_param_selected)

@pytest.mark.parametrize("num, output", randomized_parameters)
def test_multiplication_11(num, output):
   assert 11*num == output
```

**Reply by RonnyPfannschmidt** [reply]
This is strongly discouraged as it tactically breaks safe parallel testing

**Reply by arthuRHD** [reply]
You're right, thanks. I'm not sure about @wang1150297847 's needs then.

**Reply by wang1150297847** [reply]
Thank you, I have found a solution。

@pytest.mark.parametrize("case_lel,id", [('P0',0),('P1',1),('P2',2)],ids=[('P0'),('P1'),('P2')])
#demo.py
class TestClass_A:
def test_demo(self,case_lel,id):
print(id)
where you use '--co',eg:
pytest --co demo.py::TestClass_A::test_demo

you can get:
<Package demo>
  <Module demo1.py>
    <Class TestClass_A>
        <Function test_demo[P0]>
        <Function test_demo[P1]>
        <Function test_demo[P2]>

so,if you just wan to run one of the parameters, you can do like this:
pytest demo.py::TestClass_A::test_demo[P0]

**Comment by RonnyPfannschmidt** [comment]
Please outline why you want to limit that specific parameter so a fitting suggestion can be done
